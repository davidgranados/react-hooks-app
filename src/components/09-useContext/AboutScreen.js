import React, { useContext } from "react";
import { AppContext } from "./AppContext";

export const AboutScreen = () => {
  const { appContext, setAppContext } = useContext(AppContext);

  const handleClick = () => {
    setAppContext({});
  };

  return (
    <div>
      <h1>AboutScreen</h1>
      <hr />

      <pre>{JSON.stringify(appContext, null, 3)}</pre>

      <button className="btn btn-warning" onClick={handleClick}>
        Logout
      </button>
    </div>
  );
};
