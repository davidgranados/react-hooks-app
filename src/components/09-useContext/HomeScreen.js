import React, { useContext } from "react";
import { AppContext } from "./AppContext";

export const HomeScreen = () => {
  const { appContext } = useContext(AppContext);

  // console.log( appContext );

  return (
    <div>
      <h1>HomeScreen</h1>
      <hr />

      <pre>{JSON.stringify(appContext, null, 3)}</pre>
    </div>
  );
};
