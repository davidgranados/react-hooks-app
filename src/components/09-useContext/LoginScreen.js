import React, { useContext } from "react";
import { AppContext } from "./AppContext";
import { useForm } from "../../hooks/useForm";

export const LoginScreen = () => {
  const { setAppContext } = useContext(AppContext);
  const [formValues, handleInputChange, formReset] = useForm({
    name: "",
    password: "",
  });

  const { name, password } = formValues;

  const handleSubmit = (e) => {
    e.preventDefault();

    setAppContext((prevContext) => {
      return {
        ...prevContext,
        user: {
          id: 123,
          name,
        },
      };
    });

    formReset();
  };
  return (
    <div>
      <h1>LoginScreen</h1>
      <hr />
      <form onSubmit={handleSubmit}>
        <h1>FormWithCustomHook</h1>
        <hr />

        <div className="form-group">
          <input
            type="text"
            name="name"
            className="form-control"
            placeholder="Tu nombre"
            autoComplete="off"
            value={name}
            onChange={handleInputChange}
          />
        </div>

        <div className="form-group">
          <input
            type="password"
            name="password"
            className="form-control"
            value={password}
            onChange={handleInputChange}
          />
        </div>

        <button type="submit" className="btn btn-primary">
          Login
        </button>
      </form>
    </div>
  );
};
