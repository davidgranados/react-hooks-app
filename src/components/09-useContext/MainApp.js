import React, { useState } from "react";
import { AppRouter } from "./AppRouter";
import { AppContext } from "./AppContext";

export const MainApp = () => {
  const [appContext, setAppContext] = useState({});

  return (
    <AppContext.Provider
      value={{
        appContext,
        setAppContext,
      }}
    >
      <AppRouter />
    </AppContext.Provider>
  );
};
