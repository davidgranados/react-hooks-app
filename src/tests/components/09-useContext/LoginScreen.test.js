import React from "react";
import { mount } from "enzyme";
import { AppContext } from "../../../components/09-useContext/AppContext";
import { LoginScreen } from "../../../components/09-useContext/LoginScreen";

describe("Pruebas en <LoginScreen />", () => {
  const setAppContext = jest.fn();

  const wrapper = mount(
    <AppContext.Provider
      value={{
        setAppContext,
      }}
    >
      <LoginScreen />
    </AppContext.Provider>
  );

  it("debe de mostrarse correctamente", () => {
    expect(wrapper).toMatchSnapshot();
  });

  it("debe de ejecutar el setAppContext", () => {
    const submitEvent = {
      preventDefault(){}
    };
    wrapper.find("form").simulate("submit", submitEvent);
    expect(setAppContext).toHaveBeenCalledTimes(1);
  });
});
