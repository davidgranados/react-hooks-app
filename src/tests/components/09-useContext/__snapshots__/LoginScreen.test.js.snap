// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <LoginScreen /> debe de mostrarse correctamente 1`] = `
<div>
  <h1>
    LoginScreen
  </h1>
  <hr />
  <form
    onSubmit={[Function]}
  >
    <h1>
      FormWithCustomHook
    </h1>
    <hr />
    <div
      className="form-group"
    >
      <input
        autoComplete="off"
        className="form-control"
        name="name"
        onChange={[Function]}
        placeholder="Tu nombre"
        type="text"
        value=""
      />
    </div>
    <div
      className="form-group"
    >
      <input
        className="form-control"
        name="password"
        onChange={[Function]}
        type="password"
        value=""
      />
    </div>
    <button
      className="btn btn-primary"
      type="submit"
    >
      Login
    </button>
  </form>
</div>
`;
